File: Array&String/HashTable/CPP/find_lucky_integer_in_an_array.cpp
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    int findLucky(vector<int>& arr) {
        int luckynum = -1;
        map<int, int> cnt;
        for (int x: arr) cnt[x]++;
        for (auto const [k, v] : cnt) if (k == v) luckynum = max(luckynum, k);
        return luckynum;
    }
};

File: Array&String/HashTable/find_lucky_integer_in_an_array.py
from collections import defaultdict
class Solution:
    def findLucky(self, arr: list[int]) -> int:
        n, cnt = -1, defaultdict(int)
        for x in arr:
            cnt[x] += 1
        for k in cnt:
            if k == cnt[k]: n = max(n, k)
        return n
